### PR TITLE
Enable hive and various things

### PR DESCRIPTION
**Pull request by cin**
added `enable-hive` conf
upgraded dependency versions
  moved scalaVersion out so it applied to all sub-projects
  fixed issue with scala 2.10 version of assembly jar being built
various other style and FP modifications
added unit tests for GeneralFunctions and SparkFuncs

### CODE REVIEW

File: cli/src/test/scala/com/ibm/sparktc/sparkbench/datageneration/LinearRegDataGenTest.scala [hunk 1]
```diff
@@ -64,7 +64,7 @@ class LinearRegDataGenTest extends FlatSpec with Matchers with BeforeAndAfterEac
 
   //TODO ultimately fix LabeledPoints being output to CSV. Surely there's a way...
   it should "throw an exception when somebody tries to output to CSV" in {
-    a [SparkBenchException] should be thrownBy {
+    a[SparkBenchException] should be thrownBy {
```
Comment (showermat) [inline cli/src/test/scala/com/ibm/sparktc/sparkbench/datageneration/LinearRegDataGenTest.scala]: If you're going to be pedantic about formatting, know that Emily wrote it the canonical way.

File: project/Dependencies.scala [hunk 2]
```diff
@@ -33,16 +33,18 @@ object Dependencies {
...
 
   val breezeDeps = Seq(
-    "org.scalanlp" % "breeze_2.11" % "0.13.2"
+    "org.scalanlp" %% "breeze" % "0.13.2"
   )
 
   val jsonCreation = Seq(
-    "com.softwaremill.sttp" %% "json4s" % "1.1.3",
-    "com.softwaremill.sttp" %% "core" % "1.1.3"
+    //    "io.spray" % "spray-json_2.11" % "1.3.4",
```
Comment (showermat) [inline project/Dependencies.scala]: Any reason not to remove this line?

File: spark-launch/src/main/scala/com/ibm/sparktc/sparkbench/sparklaunch/submission/sparksubmit/SparkSubmit.scala [hunk 3]
```diff
@@ -45,7 +46,7 @@ object SparkSubmit extends Submitter {
 
   private def submit(strSeq: Seq[String], sparkHome: String): Unit = {
     val process = Process(strSeq, None, "SPARK_HOME" -> sparkHome)
-    println(" *** SPARK-SUBMIT: " + process.toString)
+    log.error(" *** SPARK-SUBMIT: " + process.toString)
```
Comment (showermat) [inline spark-launch/src/main/scala/com/ibm/sparktc/sparkbench/sparklaunch/submission/sparksubmit/SparkSubmit.scala]: This isn't an error, is it?

**Comment by ecurtin** [comment]
LGTM except for changing the version to RELEASE

**Pull request merged**